src/Store/Features.re

```diff
@@ -561,10 +561,10 @@ let update =
   | FilesDropped({paths}) =>
     let eff =
       Service_OS.Effect.statMultiple(paths, (path, stats) =>
-        if (stats.st_kind == S_REG) {
-          OpenFileByPath(path, None, None);
-        } else {
-          Noop;
+        switch (stats.st_kind) {
+        | S_REG => OpenFileByPath(path, None, None)
+        | S_DIR => DirectoryChanged(path)
```

Comment:
One thing I'd expect in this case is that we'd `chdir` to change the working directory of the process, too. Dispatching the `DirectoryChanged` event tells our internal state the directory is changed, but doesn't actually change the working directory of the process.

We'd probably want to use `Luv.Path.chdir` to change the work directory, and then - https://github.com/aantron/luv/blob/bb11498de65a2acb56a7269801c1d2ba41672179/src/path.mli#L18 - if that is successful, dispatch the `DirectoryChanged` event.

Reply:
If the working directory isn't in sync - our file explorer and stuff will show the 'correct' directory, but when running vim commands like `:e some/file` it would be relative to the previous directory.

Reply:
That's a great point, I hadnt thought of that. Thanks! Will fix now.

Reply:
Just fixed in fe2985c!